test/e2e_node/lifecycle_hook_test.go

```diff
@@ -94,64 +99,112 @@ var _ = framework.KubeDescribe("Container Lifecycle Hook", func() {
 						Exec: &api.ExecAction{Command: []string{"touch", file}},
 					},
 				}
-				testPodWithHook(podWithHook)
+				testPodWithExecHook(podWithHook)
 			})
 		})
 
 		Context("when it is http hook", func() {
 			var targetIP string
+			var podHandleHookRequest *api.Pod
 			BeforeEach(func() {
-				By("cleanup the container to handle the HTTPGet hook request.")
-				podHandleHookRequest := getLifecycleHookTestPod("pod-handle-http-request",
-					[]string{"sh", "-c",
-						// Create test file when receive request on 1234.
-						fmt.Sprintf("echo -e \"HTTP/1.1 200 OK\n\" | nc -l -p 1234; touch %s", file),
+				By("create the container to handle the HTTPGet hook request.")
+				podHandleHookRequest = &api.Pod{
+					ObjectMeta: api.ObjectMeta{
+						Name: "pod-handle-http-request",
 					},
-				)
-				podHandleHookRequest.Spec.Containers[0].Ports = []api.ContainerPort{
-					{
-						ContainerPort: 1234,
-						Protocol:      api.ProtocolTCP,
+					Spec: api.PodSpec{
+						Containers: []api.Container{
+							{
+								Name:  "pod-handle-http-request",
+								Image: "gcr.io/google_containers/netexec:1.7",
+								Ports: []api.ContainerPort{
+									{
+										ContainerPort: 8080,
+										Protocol:      api.ProtocolTCP,
+									},
+								},
+							},
+						},
 					},
 				}
 				podHandleHookRequest = podClient.CreateSync(podHandleHookRequest)
 				targetIP = podHandleHookRequest.Status.PodIP
 			})
+			testPodWithHttpHook := func(podWithHook *api.Pod) {
+				By("create the pod with lifecycle hook")
+				podClient.CreateSync(podWithHook)
+				if podWithHook.Spec.Containers[0].Lifecycle.PostStart != nil {
+					By("check poststart hook")
+					Eventually(func() error {
+						return podClient.MatchContainerOutput(podHandleHookRequest.Name, podHandleHookRequest.Name,
+							`GET /echo\?msg=poststart`)
+					}, podWaitTimeout, podCheckInterval).Should(BeNil())
+				}
+				By("delete the pod with lifecycle hook")
+				podClient.DeleteSync(podWithHook.Name, api.NewDeleteOptions(15), podWaitTimeout)
+				if podWithHook.Spec.Containers[0].Lifecycle.PreStop != nil {
+					By("check prestop hook")
+					Eventually(func() error {
+						return podClient.MatchContainerOutput(podHandleHookRequest.Name, podHandleHookRequest.Name,
+							`GET /echo\?msg=prestop`)
+					}, podWaitTimeout, podCheckInterval).Should(BeNil())
```

Comment:
Does `DeleteSync` block till the pod has been deleted? If yes, checking the prestop hook should not need such a high timeout.